rbi/core/enumerable.rbi

```diff
@@ -1882,17 +1882,17 @@ module Enumerable
     .returns(T.untyped)
   end
   sig do
-    params(
-        initial: T.untyped,
-        blk: T.proc.params(arg0: T.untyped, arg1: Elem).returns(T.untyped),
+    type_parameters(:U).params(
+        initial: T.type_parameter(:U),
+        blk: T.proc.params(arg0: T.type_parameter(:U), arg1: Elem).returns(T.type_parameter(:U)),
     )
-    .returns(T.untyped)
+    .returns(T.type_parameter(:U))
```

Comment:
I get that this is correct as far as Sorbet goes, but I do wonder how many people will be surprised that the initial value needs to be specified as a union of the block return value and the element type, rather than Sorbet figuring that out on its own.

Reply:
Yeah, it's unclear. I guess we could ask about it?